dashboard/src/components/DeploymentFormBody/BasicDeploymentForm/Param.tsx

```diff
@@ -72,7 +72,13 @@ export default function Param({
     paramDeploymentEvent?: DeploymentEvent,
   ): boolean => {
     if (paramDeploymentEvent == null) {
-      return getValue(appValues, path) === (expectedValue ?? true);
+      let val = getValue(appValues, path);
+      // no value is provided, but there is a default one in the schema (hidden.value)
+      // https://github.com/kubeapps/kubeapps/issues/1913
+      if (!val && typeof param.hidden === "object") {
+        val = param.hidden?.value;
+      }
```

Comment:
`hidden.value` is not the default value, it's the value that the property pointed by `path` should have to be hidden. Here you are saying that the value of the property is whatever makes it hidden so it will always be true :) 

I was wrong in the issue description because I thought that `param` was the parameter pointed by `path` but it's not. You need to retrieve that `param`. To do so, you need to modify the properties of this component and include the rest of the parameters, that way you can look for it. I tried this code and it works:

```js
  const evalCondition = (
    path: string,
    expectedValue?: any,
    paramDeploymentEvent?: DeploymentEvent,
  ): boolean => {
    if (paramDeploymentEvent == null) {
      let val = getValue(appValues, path);
      if (val === undefined) {
        const target = otherParams.find(p => p.path === path);
        val = target?.value;
      }
      return val === (expectedValue ?? true);
    } else {
      return paramDeploymentEvent === deploymentEvent;
    }
  };
```

Also note the condition, I am using `val === undefined` rather than `!val` since for example `val = false` is a valid value.

Reply:
Mmm, ok, now I see... I misunderstood what `hidden.value` was actually doing. Thanks for the clarification!

>  Also you will need to write some tests, the tests for the Param component are in the BasicDeploymentForm.test.tsx file (for historical reasons :)).

I looked for a `Param.test.tsx` and didn't find anything; thanks for letting me know :). Adding a new test case.

> I am using val === undefined rather than !val ...

I thought val=false wasn't valid, but now, having a better understanding of what `hidden.value` does, I know it was.